tests/tracking/_tracking_service/test_utils.py

```diff
@@ -358,3 +361,14 @@ def test_store_object_can_be_serialized_by_pickle(tmpdir):
     pickle.dump(_get_store("https://example.com"), io.BytesIO())
     # pickle.dump(_get_store(f"sqlite:///{tmpdir.strpath}/mlflow.db"), io.BytesIO())
     # This throws `AttributeError: Can't pickle local object 'create_engine.<locals>.connect'`
+
+
+@pytest.mark.parametrize("absolute", [True, False], ids=["absolute", "relative"])
+def test_set_tracking_uri_with_path(tmp_path, monkeypatch, absolute):
```

Comment:
This test fails on windows when `absolute` is `False`:

```python
D:\a\mlflow\mlflow\mlflow\tracking\_tracking_service\utils.py:75: in set_tracking_uri
    uri = uri.resolve().as_uri()
        uri        = WindowsPath('foo/bar')
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = WindowsPath('foo/bar')

    def as_uri(self):
        """Return the path as a 'file' URI."""
        if not self.is_absolute():
>           raise ValueError("relative path can't be expressed as a file URI")
E           ValueError: relative path can't be expressed as a file URI
```

Reply:
Can we try `path = Path("foo", "bar")`?

Reply:
Weird, I call `resolve` just before `as_uri` so the path should be absolute.

Reply:
Found https://bugs.python.org/issue38671

Reply:
I'm reading https://stackoverflow.com/a/44569249 now.

Reply:
Can we call `mkdir` before `resolve`?

```python
uri.mkdir(exist_ok=True)
uri = uri.resolve().as_uri()
```

Reply:
I did some test on windows with python3.7 and `absolute` seemed to work.

I didn't knew `absolute` was deprecated tho

Reply:
Okay, I checked and mlflow create the directory anyway when we call `start_run`.

So `mkdir` should be fine.

Reply:
@cacharle Does this work on windows?

```python
(Path.cwd() / uri).resolve()
```

Test

```python
# absolute path
>>> (Path.cwd() / Path("foo").resolve()).resolve()
PosixPath('/home/haru/Desktop/repositories/mlflow/foo')
# relative path
>>> (Path.cwd() / Path("foo")).resolve()
PosixPath('/home/haru/Desktop/repositories/mlflow/foo')
```

Reply:
Works on windows aswell

Reply:
awesome, let's use this. Looks like you already pushed a commit.